router/base-http/src/main/scala/io/buoyant/router/http/ForwardClientCertFilter.scala

```diff
@@ -13,13 +13,13 @@ class ForwardClientCertFilter[Req, H: HeadersLike, Rep](implicit requestLike: Re
   val GeneralNameTypeUri = 6
   val GeneralNameTypeDns = 2
 
-  val digest: MessageDigest = MessageDigest.getInstance("SHA-256")
+  val digest = ThreadLocal.withInitial[MessageDigest](() => MessageDigest.getInstance("SHA-256"));
```

Comment:
Maybe add a comment here that explains that we use ThreadLocal because MessageDigest is not thread safe